weasyl/commishinfo.py

```diff
@@ -20,39 +52,259 @@ def convert_currency(target):
+    :param currency: The single-character currency code to convert the returned results in,
+                     as defined in commishprice.settings
+    :param offset: Results set offset for pagination. Must be a positive Number
+    :param limit: max number of results to return
+    :return: The resulting list of artists & their commission info
+    """
+    stmt = [
+        """SELECT p.userid, p.username, p.settings,
+                MIN(cp.amount_min) AS pricemin,
+                GREATEST(MAX(cp.amount_max), MAX(cp.amount_min)) AS pricemax,
+                cp.settings AS pricesettings,
+                MIN(convert.convertedmin) AS convertedmin,
+                d.content AS description, s.unixtime,
+                tag.tagcount, example.examplecount
+            FROM profile p
+
+            JOIN login on p.userid = login.userid
+
+            INNER JOIN commishclass cc ON cc.userid = p.userid
+                AND lower(cc.title) LIKE %(cclass)s
+
+            JOIN commishprice cp ON cp.classid = cc.classid
+                AND cp.userid = p.userid
+                AND cp.settings NOT LIKE '%%a'
+
+            INNER JOIN (
+                SELECT cp.priceid, cp.userid, cp.classid,
+                CASE cp.settings """
+    ]
+    # set up the cases to convert currencies from the artist's to the searcher's
+    for c in CURRENCY_CHARMAP:
+        ratio = currency_ratio(currency, c)
+        stmt.append("\n WHEN '%s' THEN MIN(cp.amount_min) / %f" % (c, ratio))
+
+    stmt.append("""
+                    ELSE MIN(cp.amount_min)
+                END AS convertedmin
+                FROM commishprice cp
+                GROUP BY cp.priceid, cp.userid, cp.classid, cp.settings
+            ) AS convert ON convert.priceid = cp.priceid
+                AND convert.userid = p.userid
+                AND convert.classid = cc.classid
+
+            INNER JOIN (
+                SELECT MAX(unixtime) as unixtime, userid
+                FROM submission
+                GROUP BY userid
+            ) AS s ON s.userid = p.userid
+
+            JOIN commishdesc d ON d.userid = p.userid
+
+            LEFT JOIN (
+                SELECT map.targetid, COUNT(tag.tagid) AS tagcount
+                FROM searchtag tag
+                JOIN searchmapartist map ON map.tagid = tag.tagid
+                WHERE tag.title = ANY(%(tags)s)
+                GROUP BY map.targetid
+            ) AS tag ON tag.targetid = p.userid
+
+            LEFT JOIN (
+                SELECT sub.userid, COUNT (sub.submitid) as examplecount
+                FROM submission sub
+                JOIN searchmapsubmit map ON map.targetid = sub.submitid
+                JOIN searchtag tag ON map.tagid = tag.tagid
+                WHERE tag.title = ANY(%(tags)s)
+                AND sub.rating <= %(rating)s
+                AND sub.settings !~ '[hf]'
+                GROUP BY sub.userid
+            ) AS example ON example.userid = p.userid
+
+            WHERE p.settings ~ '[os]..?'
```

Comment:
I don’t really remember how `profile.settings` works, but is this equivalent to `^[os]`?